### PR TITLE
feat: analytics milestone 1 - WPB-8911

### PR DESCRIPTION
**Pull request by johnxnguyen**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-8911" title="WPB-8911" target="_blank">TopicWPB-8911</a>  Countly - Milestone 1 (Setup)
  </td></table>
  <br />
 

### Issue

This is the feature branch for Analytics Milestone 1.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -93,7 +93,6 @@ excluded:
   - "wire-ios-protos/Protos/mls.pb.swift"
   - "wire-ios-protos/Protos/otr.pb.swift"
   - "wire-ios-protos/Scripts/generate_header.swift"
-  - "wire-ios-sync-engine/Support/Sourcery/generated/AutoMockable.manual.swift"
```
Comment (netbe) [inline .swiftlint.yml]: why is this removed ?

File: WireAnalytics/Package.swift [hunk 2]
```diff
@@ -8,15 +8,20 @@ let package = Package(
...
+        .package(url: "https://github.com/DataDog/dd-sdk-ios.git", exact: "2.16.0"),
+        .package(url: "https://github.com/Countly/countly-sdk-ios.git", exact: "24.4.2"),
+        .package(path: "../SourceryPlugin")
     ],
     targets: [
         .target(
             name: "WireAnalytics",
-            dependencies: resolveWireAnalyticsDependencies(),
+            dependencies: resolveWireAnalyticsDependencies() + [
+                .product(name: "Countly", package: "countly-sdk-ios")
```
Comment (netbe) [inline WireAnalytics/Package.swift:23]: I would have expect the same setup as WireDatadog, WireCountly target that is conditionally include if ENABLE_COUNTLY was in. That way on C builds this is excluded completely

File: WireAnalytics/Sources/WireAnalyticsSupport/MockAnalyticsManagerProtocol.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireAnalytics
+
+// MARK: - MockAnalyticsManagerProtocol
+
+public class MockAnalyticsManagerProtocol: AnalyticsManagerProtocol {
```
Comment (netbe) [inline WireAnalytics/Sources/WireAnalyticsSupport/MockAnalyticsManagerProtocol.swift]: This is generated via sourcery right? did we change the pattern on WireAnalytics from all mocks being inside AutoMockable.generated.swift?

File: WireAnalytics/Sources/WireAnalyticsSupport/WireAnalyticsSupport.swift [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+// This target generates mocks via 'sourcery'. It uses the plugin configured in `Package.swift`.
+// The generated mocks are processed from the sandbox directory and are not visible in the project folder:
```
Comment (netbe) [inline WireAnalytics/Sources/WireAnalyticsSupport/WireAnalyticsSupport.swift:20]: according to this, I should not see the generated mocks then

File: wire-ios-data-model/Support/Sourcery/generated/AutoMockable.manual.swift [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+import AppKit
+#endif
+
+import Combine
+import LocalAuthentication
+import WireCoreCrypto
+
+@testable import WireDataModel
+
+public class MockZMConversationMessage: NSObject, ZMConversationMessage {
```
Comment (netbe) [inline wire-ios-data-model/Support/Sourcery/generated/AutoMockable.manual.swift:39]: out of curiosity why is this mock need to be manual, maybe add a comment here why?

File: wire-ios-sync-engine/Source/SessionManager/SessionManager.swift [hunk 6]
```diff
@@ -627,23 +643,31 @@ public final class SessionManager: NSObject, SessionManagerType {
         return account
     }
 
-    private func selectInitialAccount(_ account: Account, launchOptions: LaunchOptions) {
+    private func selectInitialAccount(
```
Comment (netbe) [inline wire-ios-sync-engine/Source/SessionManager/SessionManager.swift:649]: add doc especially on completion Bool parameter

File: wire-ios-sync-engine/Support/Sourcery/generated/AutoMockable.manual.swift [hunk 7]
```diff
@@ -100,3 +102,170 @@ public class MockSessionEstablisherInterface: SessionEstablisherInterface {
     }
 
 }
+
+public class MockMessageAppendableConversation: MessageAppendableConversation {
```
Comment (netbe) [inline wire-ios-sync-engine/Support/Sourcery/generated/AutoMockable.manual.swift:106]: same here what's the reason again is it the `any` keyword?

File: wire-ios/Configuration/Debug.xcconfig [hunk 8]
```diff
@@ -31,3 +31,5 @@ PROVISIONING_PROFILE_SPECIFIER_NOTIFICATION_EXT = Wire iOS Alpha Notification Se
 
 // App icon from asset bundle
 APPICON_NAME = AppIcon-Debug
+
+COUNTLY_APP_KEY = 37cc418a6fc07403b0a918252133c15b95486883
```
Comment (netbe) [inline wire-ios/Configuration/Debug.xcconfig]: to revert ⚠️

File: wire-ios/Wire-iOS Tests/ReferenceImages/SettingsTableViewControllerSnapshotTests/testForDataUsagePermissionsForPublicDomain.1.png
Comment (netbe) [inline wire-ios/Wire-iOS Tests/ReferenceImages/SettingsTableViewControllerSnapshotTests/testForDataUsagePermissionsForPublicDomain.1.png:1]: does that need to be in git-lfs? cc @caldrian

**Review by netbe — CHANGES_REQUESTED** [review]

File: wire-ios/Wire-iOS Tests/ReferenceImages/SettingsTableViewControllerSnapshotTests/testForDataUsagePermissionsForPublicDomain.1.png
Comment (netbe) [reply wire-ios/Wire-iOS Tests/ReferenceImages/SettingsTableViewControllerSnapshotTests/testForDataUsagePermissionsForPublicDomain.1.png:1]: nevermind this is not develop

File: WireAnalytics/Package.swift (hunk 2 above)
Comment (agisilaos) [reply WireAnalytics/Package.swift:23]: We're going to tackle this next sprint as discussed.

**Review by netbe — APPROVED** [review]

File: .swiftlint.yml (hunk 1 above)
Comment (netbe) [reply .swiftlint.yml]: @agisilaos do you know? the file is still present, I would not expect generated code to be linted

Comment (agisilaos) [reply .swiftlint.yml]: I will bring it back.

Comment (agisilaos) [reply .swiftlint.yml]: Done!

**Review by agisilaos — APPROVED** [review]
